crates/viewer/re_viewer_context/src/time_control.rs

```diff
@@ -1516,136 +1553,324 @@ fn default_timeline<'a>(timelines: impl IntoIterator<Item = &'a TimelineStats>)
+
+        // Multiple timelines - log_time should win over log_tick when counts are equal
+        let (hist_both, timelines_both) = create_test_data(&[log_time, log_tick], &[42, 42]);
+        assert_eq!(default_timeline(&hist_both, &timelines_both), log_time);
+
+        // Custom timeline should win over both log_time and log_tick when counts are equal
+        let (hist_custom0, timelines_custom0) =
+            create_test_data(&[log_time, log_tick, custom_timeline0], &[42, 42, 42]);
         assert_eq!(
-            default_timeline([&log_time, &log_tick, &custom_timeline0]),
-            custom_timeline0.timeline
+            default_timeline(&hist_custom0, &timelines_custom0),
+            custom_timeline0
         );
+
+        // Order shouldn't matter
+        let (hist_custom0_rev, timelines_custom0_rev) =
+            create_test_data(&[custom_timeline0, log_time, log_tick], &[42, 42, 42]);
         assert_eq!(
-            default_timeline([&custom_timeline0, &log_time, &log_tick]),
-            custom_timeline0.timeline
+            default_timeline(&hist_custom0_rev, &timelines_custom0_rev),
+            custom_timeline0
         );
+
+        let (hist_custom0_mid, timelines_custom0_mid) =
+            create_test_data(&[log_time, custom_timeline0, log_tick], &[42, 42, 42]);
         assert_eq!(
-            default_timeline([&log_time, &custom_timeline0, &log_tick]),
-            custom_timeline0.timeline
+            default_timeline(&hist_custom0_mid, &timelines_custom0_mid),
+            custom_timeline0
         );
+
+        // Custom timelines with different counts - higher count wins
+        let (hist_custom1, timelines_custom1) =
+            create_test_data(&[custom_timeline0, custom_timeline1], &[42, 43]);
         assert_eq!(
-            default_timeline([&custom_timeline0, &log_time]),
-            custom_timeline0.timeline
+            default_timeline(&hist_custom1, &timelines_custom1),
+            custom_timeline1
         );
+
+        // Single custom timeline
+        let (hist_single_custom, timelines_single_custom) =
+            create_test_data(&[custom_timeline0], &[42]);
         assert_eq!(
-            default_timeline([&custom_timeline0, &log_tick]),
-            custom_timeline0.timeline
+            default_timeline(&hist_single_custom, &timelines_single_custom),
+            custom_timeline0
         );
+    }
+
+    #[test]
+    fn test_step_fwd_time() {
+        use re_log_types::TimeReal;
+
+        let mut hist = TimeHistogram::default();
+        hist.increment(10, 1);
+        hist.increment(20, 1);
+        hist.increment(30, 1);
+
+        // Step forward from before first key
+        assert_eq!(
+            step_fwd_time(TimeReal::from(5), &hist),
+            TimeInt::new_temporal(10)
+        );
+
+        // Step forward from middle
+        assert_eq!(
+            step_fwd_time(TimeReal::from(15), &hist),
```

Comment:
Again here: please also test what happens when we are exactly on one of the existing times (10, 20, 30)